spec.html

```diff
@@ -34532,8 +34534,203 @@ <h2>Syntax</h2>
+                <ul>
+                  <li>
+                    `[` `]` for nested character classes and end-of-class, and for lexer scanning
+                  </li>
+                  <li>
+                    `-` for ranges
+                  </li>
+                  <li>
+                    `\\` for escape sequences
+                  </li>
+                </ul>
+              </li>
+              <li>
+                Characters that must be escaped (for the JS-level lexer):
+                <ul>
+                  <li>
+                    `/` for ES regex literals, and for lexer scanning
+                  </li>
+                </ul>
+              </li>
+              <li>
+                Characters that should be escaped:
+                <ul>
+                  <li>
+                    `(` `)` reserved for future extensions, and for readability
+                  </li>
+                  <li>
+                    `{` `}` reserved for future extensions, and for readability
+                  </li>
+                  <li>
+                    `|` for consistent escaping anywhere in a character class (needed in string literals)
+                  </li>
+                </ul>
+              </li>
+            </ul>
+          </li>
+          <li>
+            FYI: Required escaping in character classes string literals:
+            <ul>
+              <li>
+                Characters that must be escaped (to make the syntax work):
+                <ul>
+                  <li>
+                    `{` `}` for end-of-string-literal, and the open brace for readability
+                  </li>
+                  <li>
+                    `\\` for escape sequences
+                  </li>
+                  <li>
+                    `|` for multiple-string literals
+                  </li>
+                </ul>
+              </li>
+              <li>
+                Characters that must be escaped (for one or another lexer):
+                <ul>
+                  <li>
+                    `[` `]` for lexer scanning
+                  </li>
+                  <li>
+                    `/` for ES regex literals, and for lexer scanning
+                  </li>
+                </ul>
+              </li>
+              <li>
+                Characters that should be escaped:
+                <ul>
+                  <li>
+                    `(` `)` reserved for future extensions, and for readability
+                  </li>
+                  <li>
+                    `-` for consistent escaping anywhere in a character class (needed in ranges)
+                  </li>
+                </ul>
+              </li>
+            </ul>
+          </li>
+        </ul>
+      </emu-note>
+      <emu-grammar type="definition">
+        ClassSetReservedPunctuator :: one of
+          `&amp;` `-` `!` `#` `%` `,`
+          `:` `;` `&lt;` `=` `&gt;` `@`
+          `_` `&grave;` `~`
```

Comment:
```suggestion
          `&grave;` `~`
```

Reply:
Moved to https://github.com/mathiasbynens/ecma262/pull/12